llvm/include/llvm/CodeGen/TargetRegisterInfo.h

```diff
@@ -46,7 +46,7 @@ class TargetRegisterClass {
 public:
   using iterator = const MCPhysReg *;
   using const_iterator = const MCPhysReg *;
-  using sc_iterator = const TargetRegisterClass* const *;
+  using sc_iterator = const unsigned *;
```

Comment:
Ideally this would be a full-blown class so as to not expose implementation details to users of `getSuperClasses` (which should return `iterator_range`).
See e.g. `MCRegisterInfo::superregs`.